tests/build-rss.test.js

```diff
@@ -0,0 +1,81 @@
+const fs = require('fs');
+const path = require('path');
+const rssFeed = require('../scripts/build-rss');
+const { mockRssData, title, type, desc, outputPath } = require('./fixtures/rssData');
+
+jest.mock('../config/posts.json', () => mockRssData, { virtual: true });
+
+describe('rssFeed', () => {
+  const testOutputDir = path.join(__dirname, '..', 'public', 'test-output');
+
+  beforeEach(() => {
+    fs.mkdirSync(testOutputDir, { recursive: true });
+  });
+
+  afterEach(() => {
+    if (fs.existsSync(testOutputDir)) {
+      fs.readdirSync(testOutputDir).forEach(file => {
+        fs.unlinkSync(path.join(testOutputDir, file));
+      });
+      fs.rmdirSync(testOutputDir);
+    }
+  });
+
+  it('should generate RSS feed and write to file', () => {
+    rssFeed(type, title, desc, outputPath);
```

Comment:
can we add a `try/catch` block here to track the err variable here and see if any error is not returned from function execution? Try to do it for each test.

Reply:
Done